runtime/src/accounts_db.rs

```diff
@@ -702,10 +707,9 @@ impl AccountsDB {
         // Calculate store counts as if everything was purged
         // Then purge if we can
         let mut store_counts: HashMap<AppendVecId, (usize, HashSet<Pubkey>)> = HashMap::new();
-        let storage = self.storage.read().unwrap();
```

Comment:
From my understanding, the only extra risk this adds here, is that where before holding this lock would prevent any other slots in `self.storage` from being modified, now other slots can be modified. The three places this can happen

1. https://github.com/solana-labs/solana/blob/master/runtime/src/accounts_db.rs#L970-L974. Shouldn't be possible b/c we hold the `shrink_candidate_slots` lock, and `shrink_all_slots() -> do_shrink_slot_forced()` is not called after startup, so the lock should be respected  

2. https://github.com/solana-labs/solana/blob/master/runtime/src/accounts_db.rs#L1256-L1259. Can this remove a slot that exists in `account_infos` such that the call below to `self.storage.0.get(&slot).unwrap();` panics?

3. https://github.com/solana-labs/solana/blob/master/runtime/src/accounts_db.rs#L1236-L1238. Adding a new storage entry should be ok as that should be on some future non-rooted slot which shouldn't exist in `account_infos`

Reply:
@carllin 

1. Yeah, I think your understanding is correct.
2. Yeah, can `panic!`. you are addressing this at #12194 
3. Yeah, this is correct as well.

Reply:
Awesome, thanks!